test/suite/test_prepare06.py

```diff
@@ -0,0 +1,94 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_prepare06.py
+#   Prepare: Rounding off prepared transactions Timestamps.
+#
+
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_prepare06(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_prepare06'
+    uri = 'table:' + tablename
+
+    def test_timestamp_api(self):
+        self.session.create(self.uri, 'key_format=i,value_format=i')
+        c = self.session.open_cursor(self.uri)
+
+        # It is illegal to set a prepare timestamp older than oldest timestamp.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(20))
+        self.session.begin_transaction()
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(10)),
+            "/older than the oldest timestamp/")
+        self.session.rollback_transaction()
+
+        # Check setting the prepared timestamps earlier to oldest timestamp is
+        # valid with roundup_timestamps settings.
+        self.session.begin_transaction('roundup_timestamps=(prepared=true)')
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(10))
+        self.session.commit_transaction('commit_timestamp=' + timestamp_str(15))
+
+        # It is illegal to set a prepare timestamp same as or earlier than an
+        # active read timestamp even with roundup_timestamps settings.
+        # Start a new reader to have an active read timestamp.
+        s_reader = self.conn.open_session()
+        s_reader.begin_transaction('read_timestamp=' + timestamp_str(40))
+        self.session.begin_transaction()
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(10)),
+            "/not later than an active read timestamp/")
```

Comment:
I would like to see both cases. The one you added is the earlier than the reader case. Please add an equal to the reader case so that if the edge case ever changes, we catch it.